beancount_import/training.py

```diff
@@ -126,8 +130,22 @@ def default_extractor(x, key_value_pairs, key=key):
                     getattr(self, t)[key] = extractor
 
     def _ignore_posting_for_automatic_classification(self, posting):
-        return re.match(self.ignore_account_pattern,
+        skip = self.skip_accounts_cache.get(posting.account)
+        if skip is not None:
+            return skip
+
+        # Cache because checking prefixes can be an expensive O(m*n)
+        # operation.
+        for parent in self.skip_accounts:
```

Comment:
I like the cache.  You could additionally possibly make the matching more efficient by dynamically constructing a regular expression from the set.  But I think it is fine as is.

Reply:
I like that idea. Done. I've also switched to functools.lru_cache to make this simpler.